django/db/models/functions/window.py

```diff
@@ -0,0 +1,119 @@
+from django.db.models import FloatField, Func, IntegerField
+
+__all__ = [
+    'CumeDist', 'DenseRank', 'FirstValue', 'Lag', 'LastValue', 'Lead',
+    'NthValue', 'Ntile', 'PercentRank', 'Rank', 'RowNumber',
+]
+
+
+class WindowFunction(Func):
+    window_compatible = True
+
+
+class CumeDist(WindowFunction):
+    function = 'CUME_DIST'
+    name = 'CumeDist'
+
+    def __init__(self, *expressions, **extra):
+        super().__init__(*expressions, output_field=FloatField())
```

Comment:
`ouput_field` should work just fine, the issue is really that `runtests.py` should follow our documented guidelines and make sure to call `django.setup()` before attempting any other `django` imports. Here's the stacktrace @atombrella mentioned (triggered by setting `django.db.models.functions.base.Now.output_field = fields.DateTimeField()` and trying to run tests).

```
Traceback (most recent call last):
  File "./runtests.py", line 17, in <module>
    from django.test import TestCase, TransactionTestCase
  File "/Users/charettes/Workspace/django/django/test/__init__.py", line 3, in <module>
    from django.test.client import Client, RequestFactory
  File "/Users/charettes/Workspace/django/django/test/client.py", line 12, in <module>
    from django.core.handlers.base import BaseHandler
  File "/Users/charettes/Workspace/django/django/core/handlers/base.py", line 7, in <module>
    from django.urls import get_resolver, set_urlconf
  File "/Users/charettes/Workspace/django/django/urls/__init__.py", line 1, in <module>
    from .base import (
  File "/Users/charettes/Workspace/django/django/urls/base.py", line 9, in <module>
    from .exceptions import NoReverseMatch, Resolver404
  File "/Users/charettes/Workspace/django/django/urls/exceptions.py", line 1, in <module>
    from django.http import Http404
  File "/Users/charettes/Workspace/django/django/http/__init__.py", line 5, in <module>
    from django.http.response import (
  File "/Users/charettes/Workspace/django/django/http/response.py", line 14, in <module>
    from django.core.serializers.json import DjangoJSONEncoder
  File "/Users/charettes/Workspace/django/django/core/serializers/__init__.py", line 23, in <module>
    from django.core.serializers.base import SerializerDoesNotExist
  File "/Users/charettes/Workspace/django/django/core/serializers/base.py", line 6, in <module>
    from django.db import models
  File "/Users/charettes/Workspace/django/django/db/models/__init__.py", line 19, in <module>
    from django.db.models.manager import Manager
  File "/Users/charettes/Workspace/django/django/db/models/manager.py", line 6, in <module>
    from django.db.models.query import QuerySet
  File "/Users/charettes/Workspace/django/django/db/models/query.py", line 23, in <module>
    from django.db.models.functions import Trunc
  File "/Users/charettes/Workspace/django/django/db/models/functions/__init__.py", line 1, in <module>
    from .base import (
  File "/Users/charettes/Workspace/django/django/db/models/functions/base.py", line 161, in <module>
    class Now(Func):
  File "/Users/charettes/Workspace/django/django/db/models/functions/base.py", line 163, in Now
    output_field = fields.DateTimeField()
  File "/Users/charettes/Workspace/django/django/db/models/fields/__init__.py", line 1161, in __init__
    super().__init__(verbose_name, name, **kwargs)
  File "/Users/charettes/Workspace/django/django/db/models/fields/__init__.py", line 161, in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
  File "/Users/charettes/Workspace/django/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/Users/charettes/Workspace/django/django/conf/__init__.py", line 41, in _setup
    % (desc, ENVIRONMENT_VARIABLE))
django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
```